client/my-sites/upgrades/domain-management/name-servers/dns-template-selector.jsx

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import SegmentedControl from 'components/segmented-control';
+
+class DnsTemplateSelector extends Component {
+	handleOnSelect = ( option ) => {
+		this.props.onTemplateClick( option.label );
+	}
+
+	render() {
+		const { templates } = this.props;
+		return (
+			<SegmentedControl
+				primary={ true }
+				options={
+					templates.map( ( template ) => {
+						return {
+							value: template.dnsTemplate,
+							label: template.name
+						};
+					}, this )
```

Comment:
I don't think `this` is needed (ha-ha, get it? 🤓) - the arrow function takes care of that.